src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/PropertyExtensions.cs

```diff
@@ -30,5 +30,27 @@ public static async Task<Guid> GetValueAsGuidAsync(this IProperty property)
 
             return Guid.Empty;
         }
+
+        /// <summary>
+        ///     Returns the value of the specific <see cref="IProperty"/> as an enumeration, or
+        ///     the default of <typeparamref name="TEnum"/> if it cannot be parsed.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="property"/> is <see langword="null"/>.
+        /// </exception>
+        public static async Task<TEnum> GetValueAsEnumAsync<TEnum>(this IProperty property)
+            where TEnum : struct
```

Comment:
You can use an enum type constraint here.

Reply:
Tried that first, unfortunately the parse isn't `T : enum`

Reply:
I think it should work:

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAcoKgW1QOzgBsiBjACwlIGsBLfAcwBoYRaimATEAagB8AAgCYAjAFgAUAIDMAAmGyAwrIDek2RtkRCOWQFFVsgL7rNM+QBZZAWQA8AFQB8ACgCUsgO6Uks+7JCyyDAIcKQwTPo6qqaamno6AHT2CACeAApYyBDOAEQAgjkRtAz4APZIilBZAcFwEBGlcDC+skjIxDCuANwxxpJGQA==

Reply:
Ah, didn't think to put struct in there as well. Thanks!

Reply:
Yeah it seems redundant.